File: examples/python/basic.py
"""
Minimal Python control envelope creation.
"""
from dataclasses import dataclass, asdict
import json
import time
import os


@dataclass
class ControlHeader:
    seq: int
    nonce: bytes
    timestamp_ms: int


header = ControlHeader(seq=1, nonce=b"\x01\x02\x03", timestamp_ms=int(time.time() * 1000))
payload = {"type": "GetStatus"}

envelope = {"header": asdict(header), "payload": payload, "signature": ""}

print(json.dumps(envelope, indent=2))

File: examples/ts/basic.ts
import {
  ControlEnvelope,
  ControlHeader,
  ControlPayload,
  SetMode,
} from "../../bindings/ts/src";

const header: ControlHeader = {
  seq: 1,
  nonce: new Uint8Array([1, 2, 3]),
  timestamp_ms: Date.now(),
};

const payload: ControlPayload = {
  type: "SetMode",
  body: { mode: "Normal" },
};

const envelope: ControlEnvelope = {
  header,
  payload,
  signature: new Uint8Array([]), // attach Ed25519 signature here
};

console.log("Example envelope", envelope);
